### PR TITLE
A11y contribution

### PR DESCRIPTION
**Pull request by prshelke**
### Description of Change ###

Added a new property in the SemanticEffects to control the accessibility visibility of descendants of a control. 

### Bugs Fixed ###
This property can be used on any control especially Grid or Stacklayout to hide/show its descendants from accessibility.

- Fixes #

### API Changes ###

Added: 
 
- `enum ImportantForA11y`

Changed:

None

### Behavioral Changes ###

None

### PR Checklist ###

- [ ] Has a linked Issue, and the Issue has been `approved`
- [ ] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard

- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Comment by jfversluis** [comment]
Thanks for this @prshelke!

I would like to avoid using the term "a11y" in code though, would you mind refactoring this to accessibility?

For the rest @rachelkang ?

**Comment by prshelke** [comment]
@jfversluis Sure! I have signed the CLA. I will also make the required changes as suggested.

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Semantic/SemanticEffectRouter.ios.cs [hunk 1]
```diff
@@ -26,6 +27,23 @@ protected override void Update(UIView view, SemanticEffectRouter effect)
 			else
 				view.AccessibilityTraits &= ~UIAccessibilityTrait.Header;
 
+			var a11yVisibility = SemanticEffect.GetA11yVisibility(Element);
+			switch (a11yVisibility)
+			{
+				case ImportantForA11y.Auto:
+					view.AccessibilityElementsHidden = false;
```
Comment (PureWeen) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Semantic/SemanticEffectRouter.ios.cs:34]: I'm pretty sure the logic here for going back to Auto will also need to make sure it resets the `IsAccessibilityElement` property

Which you should be able to do like this

```C#
view.IsAccessibilityElement = view is UIControl
```

**Review by rachelkang — DISMISSED** [review]
After discussing various options with @PureWeen, I've updated the name of the properties and values to more accurately reflect the property's behavior.

`SemanticInclusion` with values `Default`, `Include`, `Exclude`, `ExcludeWithChildren`

This can be merged after this PR https://github.com/xamarin/XamarinCommunityToolkit/pull/1381 gets merged

**Comment by DDSoftDennie** [comment]
About naming: what about Inclusive or ImportantForInclusion? A11y is not a great practice I guess but AccessibilitySomeProperty is quite long.

**Review by rachelkang — APPROVED** [review]
LGTM! @jfversluis

**Pull request merged**